src/Symfony/Component/Form/Tests/Extension/Validator/DataTransformer/ViewObjectTransformer.php

```diff
@@ -0,0 +1,73 @@
+<?php
+namespace Symfony\Component\Form\Tests\Extension\Validator\DataTransformer;
+
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+
+/**
+* Transforms Object to array for view
+*
+* @author Charles J. C. Elling <charles@denumeris.com>
+*/
+class ViewObjectTransformer implements DataTransformerInterface {
```

Comment:
Yeah but that's a really specific case were model can be a string or an object, and even when the model data is a string the normalized data is always a `\DateTime` object.

If your model is always an object so should (not must) your view data.

>1. Any form fields that are not readable in the object, like the discriminator for the entity hierarchy that is stored in the entity metadata.
>2. Any form fields that are not writable in the object, like the id that usually only has a getter method.

Ok but why such fields would be present? Usually the `id` does not have a setter and does not have a field either, and even if it has a field, one can use the `mapped` option set to `false` so the data mapper does not call the setter.

I hope your working example will help us to find a way to solve this issue.

Your change looks weird as you are duplicating the logic of  https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/Form.php#L215 which is called [a few lines above]( https://github.com/symfony/symfony/pull/20641/files#diff-7b14a00bf598c6870d7e3556f4bb4ff5R191).

But it's ok, I can be wrong and your change legit, let's just try to investigate further so we can be sure before merging your PR :)